update release note generation

radTuti opened the pull request:
## Description

- ensure milestone is closed before generating release notes
- ensure there are no open PRs in the milestone

## For PR author

- [ ] Tests for change.
- [ ] If changing pkg/apis/, run `make gen-files`
- [ ] If changing versions, run `make gen-versions`

## For PR reviewers

A note for code reviewers - all pull requests must have the following:

- [ ] Milestone set according to targeted release.
- [ ] Appropriate labels:
  - `kind/bug` if this is a bugfix.
  - `kind/enhancement` if this is a a new feature.
  - `enterprise` if this PR applies to Calico Enterprise only.

radTuti commented on hack/generate_release_notes.py:
```diff
@@ -47,27 +47,37 @@ def issues_in_milestone() -> list:
+        )
+    # Since we only want merged PRs, raise an error if there are none.
+    if len(merged_issues) == 0:
+        raise ReleaseNoteError(f"no issues found for milestone {m.title}")
```
In that case, we should not be using this tooling to generate release notes.

danudey replied:
It still generates the outline for the release notes, so it's still valuable to run.

tmjd commented on RELEASING.md:
````diff
@@ -95,64 +101,23 @@ Go to the PR created and:
-```
+    ```sh
+    git tag v1.30.3
+    git push --tags
````
I'd suggest pushing only the newly created tag. If a user has other tags locally we probably don't want them all pushed.

caseydavenport replied:
Yep, 100% this.

tmjd commented on RELEASING.md:
```diff
@@ -95,64 +101,23 @@ Go to the PR created and:
-1. Go to https://github.com/tigera/operator/releases and edit the release tag to include the generated release notes, and update the title.
+1. Publish the release.
 
-1. Close the milestone for this release. https://github.com/tigera/operator/milestones
```
This still must be done AFAICT, don't we need to move this before doing the 'Rerun' step above or else the Rerun won't create the release draft?

radTuti replied:
The release notes now require that the milestone is closed first before generating the draft release with release notes.

tmjd commented:
Seems like these checks on the milestone and PRs in the milestone should perhaps be done as part of the `release-prep` or there should be a check before doing the "Rerun" in semaphore.

radTuti commented:
> Seems like these checks on the milestone and PRs in the milestone should perhaps be done as part of the `release-prep` also or there should be a check before doing the "Rerun" in semaphore.

I have moved the checking milestones and PRs in milestones to "Preparing for the release" section in the releasing docs. The check in the release notes are additional sanity check.
I have also added to the `release-prep` task to handle checking the milestone

tmjd commented on Makefile:
```diff
@@ -565,7 +565,19 @@ ifdef LOCAL_BUILD
+	@gh search prs --milestone $(VERSION) --state open --json id | \
+		jq 'if . == [] then empty else error("there are open PRs in milestone") end'
+	$(eval MILESTONE_NUMBER := $(shell gh milestone list --query $(VERSION) --repo $(REPO) --state closed --json number --jq '.[0].number'))
+	@echo "Closing $(VERSION) milestone #$(MILESTONE_NUMBER)"
```
You've already got in the readme to close the Milestone, I don't think we should do that here also, I think here we should just check that the milestone is closed.

radTuti replied:
updated to check that the milestone is closed

tmjd reviewed (approved):
LGTM
